dashboard/test/models/activity_test.rb

```diff
@@ -1,5 +1,6 @@
 require 'test_helper'
 
+require 'async_progress_handler'
```

Comment:
While I personally think IWYU is good engineering discipline to follow, it's generally discouraged by the Rails framework so I'd recommend against it within code that's part of a Rails application proper (for `code-dot-org` that would be the stuff in the `/dashboard` folder).

More specifically, the ["Autoloading and `require`"](http://guides.rubyonrails.org/autoloading_and_reloading_constants.html#autoloading-and-require) section of the [Autoloading and Reloading Constants](http://guides.rubyonrails.org/autoloading_and_reloading_constants.html) says that Rails-autoloaded constants should never be explicitly `require`d.

However I don't think the `AsyncProgressHandler` needs to be Rails-application-specific anyway, so it might make more sense to eventually extract it into a third-party-library folder.